File: strings/src/test/java/org/example/strings/URLifyTest.java
package org.example.strings;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.assertj.core.api.Assertions.assertThat;

public class URLifyTest {
    
    @Test
    public void containsNoSpaces() {
        testReplaceSpaces("Coding", "Coding");
    }

    @Test
    @Disabled
    public void containsSingleSpace() {
        testReplaceSpaces("Canton Coders", "Canton%20Coders");
    }

    @Test
    @Disabled
    public void containsMultipleSpaces() {
        testReplaceSpaces("Coding with Canton Coders", "Coding%20with%20Canton%20Coders");
    }

    private static void testReplaceSpaces(String input, String output) {
        var inputArray = Arrays.copyOf(input.toCharArray(), output.length());
        Arrays.fill(inputArray, input.length(), output.length(), ' ');

        Urlify.replaceSpaces(inputArray);
        assertThat(inputArray).isEqualTo(output.toCharArray());
    }
}

File: strings/csharp/StringsTest/URLifyTest.cs
using Newtonsoft.Json.Linq;
using NUnit.Framework;
using StringHomework;
using System;
using System.Collections.Generic;
using System.Text;

namespace StringHomeworkTest
{
    internal class URLifyTest
    {


        [Test]        
        public void ContainsNoSpaces()
        {
            TestReplaceSpaces("Coding", "Coding");
        }

        [Test]
        public void ContainsSingleSpaceContainsNoSpaces()
        {
            TestReplaceSpaces("Canton Coders ", "Canton%20Coders");
        }

        [Test]
        public void ContainsMultipleSpaces()
        {
            TestReplaceSpaces("Coding with Canton Coders", "Coding%20with%20Canton%20Coders");
        }

        private static void TestReplaceSpaces(String input, String output)
        {
            var inputArray = new char[output.Length];
            Array.Copy(sourceArray: input.ToCharArray(),
                       inputArray,
                       output.Length);

            URLify.ReplaceSpaces(inputArray);
            Assert.AreEqual(inputArray, output.ToCharArray());
        }

    }
}
